docs/source/packs.rst

```diff
@@ -210,6 +210,163 @@ have to comply with Health and Safety. The generated file will be placed in
 
 For more nice tricks on pack configuration, see :doc:`/reference/pack_configs`.
 
+Overriding Pack Defaults
+------------------------
+
+When installing a pack the state of the resources is taken from the metadata files in the pack. Occassionally, when installing community packs you might not want all resources enabled, e.g. you only want to use a subset of actions, or want to disable the sensors.
+
+Prior, to |st2| 3.7.0 then this state can be altered, by:
+
+  * using the |st2| APIs to disable the resource. However, this will be forgotten upon a ``st2ctl reload`` or pack reinstall.
+  * changing the metadata file manually. This would be lost upon an upgrade, and cannot be easily tracked.
+
+In |st2| 3.7.0 we have introduced the override feature, so that the metadata of a packs resources can be overridden by configuration files. This will always be read upon a reload or pack install. The ST2 APIs will still allow you to override this, but as before any changes made by the |st2| APIs to enable/disable resources is forgotten upon reload or re-install.
+
+The override facility is currently restricted to allowing the enabled property to be overridden only for resources of type: action, alias, rule and sensor. It is controlled by the ``/opt/stackstorm/overrides`` directory.
+
+Upon a pack install or reload resource state is managed as follows:
+
+* State is read from the pack's resource metadata files that reside in ``/opt/stackstorm/packs/<packname>``. These are downloaded from the relevant repository of the pack, e.g. GIT, StackStorm-Exchange.
+* If ``/opt/stackstorm/overrides/_global.yaml`` is present, then any global overrides are applied. The ``_global.yaml`` allows you to specify the default state of a particular resource types, e.g. disable all sensors.
+  The format of the _global.yaml is as follows (set to disable everything):
+
+.. code-block:: bash
+
+   ---
+   sensors:
+     defaults:
+       enabled: false
+   actions:
+     defaults:
+       enabled: false
+   aliases:
+     defaults:
+       enabled: false
+   rules:
+     defaults:
+       enabled: false
+
+* If ``/opt/stackstorm/overrides/<packname>.yaml`` is present, then any pack defaults, or resource specific overrides are applied. The <pack>.yaml allows you to alter the state of all the resources of a particular type, via the defaults property (e.g. all sensors within the pack), or by individual resource type, via the exceptions property (e.g. only alter state of individual actions).
+  The format of the ``<pack>.yaml`` is as follows (set to disable everything, except for individual resource of each type):
+
+.. code-block:: bash
+
+   ---
+   sensors:
+     defaults:
+       enabled: false
+     exceptions:
+       sensorname1:
```

Comment:
I believe we disallow `<pack_name>.<resource_name>` here, right? If so, it might be nice to mention that somewhere (I'm not sure where). Something like, "Within the overrides files, pack names are implicit based on which file it is in. So, in `/opt/stackstorm/overrides/<packname>.yaml` use `<sensorname>` not `<packname>.<sensorname>`."

wdyt?

Reply:
@cognifloyd Added some more explanation to hopefully make it clearer, let me know what you think.

Reply:
Awesome. Thanks! Is this still WIP?

Reply:
Nope - just noticed that too - changed title!